Code/Source/Transform/ROS2FrameComponent.cpp

```diff
@@ -6,20 +6,57 @@
  *
  */
 #include "ROS2/ROS2Bus.h"
+#include "Transform/ROS2FrameComponent.h"
 #include "Utilities/ROS2Names.h"
 #include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
 #include <AzCore/Serialization/SerializeContext.h>
 
 namespace ROS2
 {
     void ROS2FrameComponent::Activate()
     {
-        m_transformPublisher = AZStd::make_unique<TransformPublisher>(GetParentFrameID(), GetFrameID(), GetFrameTransform());
+        if (m_publishTransform)
+        {
+            m_ros2Transform = AZStd::make_unique<ROS2Transform>(GetParentFrameID(), GetFrameID(), IsDynamic());
+            if (IsDynamic())
+            {
+                AZ::TickBus::Handler::BusConnect();
+            }
+            else
+            {
+                m_ros2Transform->Publish(GetFrameTransform());
+            }
+        }
     }
 
     void ROS2FrameComponent::Deactivate()
     {
-        m_transformPublisher.reset();
+        if (m_publishTransform)
+        {
+            if (IsDynamic())
+            {
+                AZ::TickBus::Handler::BusDisconnect();
+            }
+            m_ros2Transform.reset();
+        }
+    }
+
+    void ROS2FrameComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        m_ros2Transform->Publish(GetFrameTransform());
```

Comment:
I think we should be able to control the working frequency of the non-dynamic transform publisher. Maybe add TODO or a note that it is locked to the FPS of the simulator for the current implementation.